### PR TITLE
feat: mTLS support

### PR DESCRIPTION
**Pull request by tower-nyc**
Change-Id: I90f61317b47062cffff6af0c92db7feff2232aa4
Fixes for #65

### CODE REVIEW

File: charts/redpanda/templates/post-install-upgrade-job.yaml [hunk 1]
```diff
@@ -100,6 +103,10 @@ spec:
           {{- range $name, $cert := .Values.tls.certs }}
           - name: redpanda-{{ $name }}-cert
             mountPath: {{ printf "/etc/tls/certs/%s" $name }}
+            {{- if (include "mtls-enabled" $ | fromJson).bool }}
+          - name: redpanda-{{ $name }}-usercert-{{ $adminUser }}
+            mountPath: {{ printf "/etc/tls/user-certs/%s/%s" $name $adminUser }}
```
Comment (RafalKorepta) [inline charts/redpanda/templates/post-install-upgrade-job.yaml:108]: I don't see this path is used

File: charts/redpanda/templates/post-upgrade.yaml [hunk 2]
```diff
@@ -97,6 +101,10 @@ spec:
   {{- range $name, $cert := .Values.tls.certs }}
           - name: redpanda-{{ $name }}-cert
             mountPath: {{ printf "/etc/tls/certs/%s" $name }}
+          {{- if (include "mtls-enabled" $ | fromJson).bool }}
+          - name: redpanda-{{ $name }}-usercert-{{ $adminUser }}
+            mountPath: {{ printf "/etc/tls/user-certs/%s/%s" $name $adminUser }}
```
Comment (RafalKorepta) [inline charts/redpanda/templates/post-upgrade.yaml:106]: I don't see this path is used

File: charts/redpanda/templates/statefulset.yaml [hunk 3]
```diff
@@ -308,6 +313,10 @@ spec:
   {{- range $name, $cert := .Values.tls.certs }}
             - name: redpanda-{{ $name }}-cert
               mountPath: {{ printf "/etc/tls/certs/%s" $name }}
+          {{- if (include "mtls-enabled" $ | fromJson).bool }}
+            - name: redpanda-{{ $name }}-usercert-{{ $adminUser }}
+              mountPath: {{ printf "/etc/tls/user-certs/%s/%s" $name $adminUser }}
```
Comment (RafalKorepta) [inline charts/redpanda/templates/statefulset.yaml:318]: I don't see this path is used

File: charts/redpanda/values.schema.json [hunk 4]
```diff
@@ -237,6 +237,66 @@
...
+        "consumer_request_timeout_ms": {"type": "integer", "default": 100},
+        "consumer_request_max_bytes": {"type": "integer", "default": 1048576},
+        "consumer_session_timeout_ms": {"type": "integer", "default": 10000},
+        "consumer_rebalance_timeout_ms": {"type": "integer", "default": 2000},
+        "consumer_heartbeat_interval_ms": {"type": "integer", "default": 500},
+        "sasl_mechanism": {"type": "string"},
+        "scram_username": {"type": "string"},
+        "scram_password": {"type": "string"}
+      }
+    },
```
Comment (RafalKorepta) [inline charts/redpanda/values.schema.json:299]: Thank you for this definition. Internally we would like to not constraint users as Redpanda might change this definition in future.

**Comment by joejulian** [comment]
This is very out-of-date and _should_ be fixed now. I'm going to close this. There's a nice feature, though, around creating the list of user certificates that I've added an issue (#775) for.

**Pull request closed without merging**